src/controllers/features/limited/addressCorrespondanceSelectorController.ts

```diff
@@ -28,8 +28,10 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
 
         // set addressoption to render the page with saved data
         let addressOption = "";
-        if (acspData.correspondenceAddress !== null) {
-            if (JSON.stringify(acspData.correspondenceAddress) === JSON.stringify(acspData.businessAddress)) {
+        const applicantDetails = acspData?.applicantDetails || {};
+        if (applicantDetails?.correspondenceAddress !== null) {
```

Comment:
If you are assigning applicantDetails to {} then it will not be null so ? is not needed. But why are we assigning it to blank?